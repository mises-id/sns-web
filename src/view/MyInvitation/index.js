/*
 * @Author: lmk
 * @Date: 2021-07-16 00:15:24
 * @LastEditTime: 2022-10-09 15:04:22
 * @LastEditors: lmk
 * @Description: Airdrop page
 */
import { getReferralList, getReferralUrl } from "@/api/user";
import Navbar from "@/components/NavBar";
import {
  Button,
  Grid,
  InfiniteScroll,
  List,
  PullToRefresh,
} from "antd-mobile";
import dayjs from "dayjs";
import React, { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom/cjs/react-router-dom.min";
// import { useLocation } from "react-router-dom";
import { Toast } from "zarm";
import "./index.scss";
const MyInvitation = () => {
  const { t } = useTranslation();
  const [info, setinfo] = useState({
    airdrop_amount: 0,
    total_channel_user: 0,
  });

  const [data, setData] = useState([]);
  const [hasMore, setHasMore] = useState(true);
  const [isLoading, setIsLoading] = useState(false);
  const [page, setPage] = useState(1);
  const [error, setError] = useState(false);
  const search = new URLSearchParams(window.location.search);
  const misesId = search.get("misesId")?.includes("did:mises:")
    ? search.get("misesId").replace("did:mises:", "")
    : search.get("misesId");
  useEffect(() => {
    if (misesId) {
      getInfo(misesId);
    }
    // eslint-disable-next-line
  }, []);
  const getInfo = async (misesId) => {
    try {
      const misesInfo = await getReferralUrl({
        misesId,
        medium: "invite",
      });
      setinfo({ ...misesInfo });
    } catch (error) {
      console.log(error);
      return await Promise.reject(error);
    }
  };
  const loadMore = async () => {
    if (isLoading || error) return;
    setIsLoading(true);
    setError(false);
    return fetchData()
      .then((res) => {
        setData([...data, ...res]);
        setPage(page + 1);
        setHasMore(res.length >= pageSize);
      })
      .finally(() => {
        setIsLoading(false);
      });
  };
  const onRefresh = async () => {
    if (isLoading) return;
    setIsLoading(true);
    setError(false);
    setPage(1);
    setHasMore(true);
    fetchData(1)
      .then((res) => {
        setData([...res]);
      })
      .finally(() => {
        setIsLoading(false);
      });
  };
  const pageSize = 20;
  const fetchData = async (pageNum) => {
    try {
      const res = await getReferralList({
        misesid: misesId,
        page_size: pageSize,
        page_num: pageNum || page,
      });
      const arr = res.data.map((val, index) => {
        // airdrop_state: 0 默认，1 正在处理 ， 2 空投成功，3 空投失败
        const misesId = val.user.misesid.replace("did:mises:", "");
        const misesStr = `${misesId.slice(0, 8)}******${misesId.slice(-4)}`;
        return {
          key: index,
          misesId: misesStr,
          airdrop_state: val.airdrop_state,
          created_at: dayjs(val.created_at).format("YYYY-MM-DD HH:mm:ss"),
          airdropState: val.airdrop_state === 2 ? "Complete" : "Incomplete",
        };
      });
      return Promise.resolve(arr);
    } catch (error) {
      return Promise.reject(error);
    }
  };
  const renderView = (val, index) => {
    return (
      <div className="table-body-item" key={index}>
        <Grid columns={2}>
          <Grid.Item>
            <div className="text-center table-header">{val.misesId}</div>
          </Grid.Item>
          <Grid.Item>
            <div className={`text-center table-header table-body-last ${val.airdrop_state===2 ? 'complete' : 'incomplete'}`}>
              {val.airdropState}
            </div>
          </Grid.Item>
        </Grid>
      </div>
    );
  };
  const share = ()=>{
    console.log(`${window.location.origin}/download?misesid=${misesId}`)
    if(!navigator.share){
      Toast.show('Browser cannot share website')
      return false;
    }
    navigator.share&&navigator.share({
      text: `Join me to use Mises Browser to start your #Web3 journey!\n\n#Mises #Browser #DeFi #NFT #SocialFi`,
      url: `${window.location.origin}/download?misesid=${misesId}`,
    }).catch(err=>{
      console.log(err)
    })
  }
  const history = useHistory()
  const customBack = ()=>{
    history.replace('/home/me')
  }
  return (
    <div>
      <Navbar title={t("myInvitationPageTitle")} customBack={customBack}/>
      <div className="m-layout">
        <div className="position-relative">
          <img
            alt=""
            src="/static/images/invite_bg@2x.png"
            width="100%"
            className="invite_bg"
          />
          <div className="position-absolute">
            <div className="text">
              <p>Invite your friends to join Mises, you </p>
              <p>can get 10% of your friends' </p>
              <p>airdrop rewards</p>
            </div>
            <div className="m-flex position-relative share-btn-box">
              <Button
                color="white"
                shape="rounded"
                size="large"
                className="m-flex-1"
                onClick={share}
              >
                <span className="share-btn">Share Mises to friends</span>
                <img
                  src="/static/images/arrow@2x.png"
                  alt=""
                  className="arrow"
                />
              </Button>
              <img
                src="/static/images/airdropIcon.png"
                alt=""
                className="airdropIcon"
              />
            </div>
          </div>
        </div>
        <div className="list">
          <p className="stas">
            Number of Invitation: {info.total_channel_user}
          </p>
          <p className="stas">
            Reward: <span>{info.airdrop_amount}</span> MIS{" "}
          </p>
          <div className="table-header-item">
            <Grid columns={2}>
              <Grid.Item>
                <div className="text-center table-header">Mises ID</div>
              </Grid.Item>
              <Grid.Item>
                <div className="text-center table-header table-header-last">
                  Status
                </div>
              </Grid.Item>
            </Grid>
          </div>
          <PullToRefresh onRefresh={onRefresh}>
            {data.length > 0 && (
              <List
                style={{ "--border-top": "none", "--border-bottom": "none" }}
              >
                {data.map(renderView)}
              </List>
            )}
            <InfiniteScroll loadMore={loadMore} hasMore={hasMore} />
          </PullToRefresh>
        </div>
      </div>
    </div>
  );
};
export default MyInvitation;
