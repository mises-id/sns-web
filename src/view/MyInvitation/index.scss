.invite_bg{
  display: block;
}
.position-relative{
  position: relative;
}
.position-absolute{
  position: absolute;
  top: 33px;
  left: 16px;
  right: 10px;

}
.text{
  font-size: 20px;
  font-weight: bold;
  color: #FFFFFF;
  line-height: 27px;
  p{
    margin-bottom: 8px;
    word-break: break-word;
  }
}
.share-btn{
  font-size: 17px;
  color: var(--theme-primary);
  font-weight: bold;
}
.arrow{
  margin-left: 6px;
  vertical-align: middle;
}
.airdropIcon{
  right: 0;
  width: 89px;
  position: absolute;
  top: -75px;
  right: -10px;
}
.share-btn-box{
  margin-top: 21px;
}
.list{
  padding: 30px 16px;
  background-color: white;
}
.table-header-item{
  background: #EEEFFF;
}
.table-header{
  line-height: 40px;
  // background: #EEEFFF;
  font-size: 12px;
  font-weight: bold;
  color: #333333;
  text-align: center;
  position: relative;
}

.table-header-last{
  &::after{
    content: "";
    position:absolute;
    width: 1px;
    height: 40px;
    left: 0;
    background: #FFFFFF;
  }
}
.table-body-last{
  &::after{
    content: "";
    position:absolute;
    width: 1px;
    height: 40px;
    left: 0;
    background: #F4F4F4;
  }
}
.table-body-item{
  &:nth-child(2n){
    background-color: #F8F8F8;
  }
  &:nth-child(2n+1){
    background-color: #FCFCFC;
  }
}
.stas{
  font-size: 15px;
  font-weight: 400;
  margin-bottom: 15px;
  color: #666666;
  span{
    color: #FF9600;
  }
  
}
.complete{
  color: #50C68D;
}
.incomplete{
  color: #999999;
}