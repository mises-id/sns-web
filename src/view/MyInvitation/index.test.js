/*
 * @Author: lmk
 * @Date: 2021-07-16 10:05:53
 * @LastEditTime: 2022-06-29 10:06:08
 * @LastEditors: lmk
 * @Description: Airdrop test
 */

import React from 'react';
import {shallow } from 'enzyme';
import MyInvitation from './';
describe('MyInvitation', () => {
  it('should render correctly', () => {
    const component = shallow(<MyInvitation />)
    expect(component).toMatchSnapshot();
  });
});