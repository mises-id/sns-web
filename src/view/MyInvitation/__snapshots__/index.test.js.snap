// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyInvitation should render correctly 1`] = `
<div>
  <Navbar
    customBack={[Function]}
    title="myInvitationPageTitle"
  />
  <div
    className="m-layout"
  >
    <div
      className="position-relative"
    >
      <img
        alt=""
        className="invite_bg"
        src="/static/images/invite_bg@2x.png"
        width="100%"
      />
      <div
        className="position-absolute"
      >
        <div
          className="text"
        >
          <p>
            Invite your friends to join Mises 
          </p>
        </div>
        <div
          className="m-flex position-relative share-btn-box"
        >
          <ForwardRef
            className="m-flex-1"
            color="white"
            onClick={[Function]}
            shape="rounded"
            size="large"
          >
            <span
              className="share-btn"
            >
              Share Mises to friends
            </span>
            <img
              alt=""
              className="arrow"
              src="/static/images/arrow@2x.png"
            />
          </ForwardRef>
        </div>
      </div>
    </div>
  </div>
</div>
`;
